### PR TITLE
Add support for auto download and dynamic context manager notation

### PR DESCRIPTION
**Pull request by sfc-gh-tszerszen**
Hey @andfanilo 🤝 😄 ,

at Snowflake ❄️ we appreciate your active participation in Streamlit community 🙇 . Your helpful presence is encouraging to Streamlit community, same goes for `streamlit-lottie` package which came to Streamlit's team attention 🤓. 

We love ❤️ this package and we are happy seeing it's popularity among our community 🧑‍🤝‍🧑.

If you allow ❓ , we'd like to propose a PR into `streamlit-lottie` package and your feedback and participation will be hugely appreciated.

This PR has backward compatibility with previous `streamlit-lottie` versions and introduces new way of using it:
* Allows auto download of lottie animation if user provides URL only
```python
import streamlit as st
from streamlit_lottie import st_lottie

with st.echo():
    st_lottie("https://assets5.lottiefiles.com/packages/lf20_V9t630.json")

```
* Monkey patches Streamlit so `streamlit-lottie` can be used as `st.lottie`
```python
import streamlit as st
import streamlit_lottie

with st.echo():
    st.lottie("https://assets5.lottiefiles.com/packages/lf20_V9t630.json")

```
* Allows `streamlit-lottie` to be used within Context Manager context, so animation will be displayed as long as some async job is running
```python
import streamlit as st
import time
from streamlit_lottie import st_lottie

st_lottie("https://assets5.lottiefiles.com/packages/lf20_V9t630.json")

with st_lottie("https://assets5.lottiefiles.com/packages/lf20_V9t630.json"):
    time.sleep(5)

```
* Introduces some smaller changes, like more `typing` package usage or updated `typescript` version

I'd like to ask you 🙇 for a review and consideration of updating `streamlit-lottie` package, what's your opinion on this? ❓ 

I'm sending Kind Regards from Streamlit Team at Snowflake ❄️ and wish you the best!
Thank you the work and activity on Streamlit Community again! 🥇

### CODE REVIEW

**Comment by andfanilo** [comment]
Hey! Thanks for the PR, just wanted to let you know I saw the message, and I made a TODO note to respond ASAP 📝 
Have a nice day 😀

**Comment by sfc-gh-tszerszen** [comment]
Thank you @andfanilo, I'll look forward to your response and take care! 🙂

File: streamlit_lottie/__init__.py [hunk 1]
```diff
@@ -17,75 +26,155 @@
...
+
+    animation_data = None
+    if isinstance(animation_source, dict):
+        animation_data = animation_source
+    elif isinstance(animation_source, str):
+        try:
+            if validate_url(animation_source):
+                animation_data = download_animation_data(animation_source)
+        except ValidationFailure:
+            # Is not url try to covert it to json
```
Comment (andfanilo) [inline streamlit_lottie/__init__.py]: covert --> convert

**Review by andfanilo — APPROVED** [review]

**Comment by andfanilo** [comment]
LGTM!
Can I let you correct the very small comment error and then I can merge/deploy a new version :) thanks for the contribution!

File: streamlit_lottie/__init__.py (hunk 1 above)
Comment (sfc-gh-tszerszen) [reply streamlit_lottie/__init__.py]: It's fixed 👍 Thank you for review and noticing 🙂

**Comment by sfc-gh-tszerszen** [comment]
Thank you @andfanilo for your acceptance and review! 🙂 
The typo is fixed and thanks again for being so active within Streamlit community! ❤️

**Comment by andfanilo** [comment]
Hey @sfc-gh-tszerszen 
I merged and I'm testing it in Python 3.8.16, Streamlit 1.22. 

I'm getting `DuplicateWidgetID: There are multiple widgets with the same key='progress'.` when using as context manager:

```python
with st.lottie("https://assets5.lottiefiles.com/private_files/lf30_3ykigvxc.json", loop=True, key="progress"):
    time.sleep(5)
```

Traceback

```
File "C:\workspace\code\streamlit-lottie-demo\app.py", line 55, in <module>
    with st.lottie(lottie_progress_url, loop=True, key="progress"):
File "c:\workspace\code\streamlit-lottie\streamlit_lottie\__init__.py", line 152, in __enter__
    self.start(key=self.key or str(uuid.uuid4().hex))
File "c:\workspace\code\streamlit-lottie\streamlit_lottie\__init__.py", line 138, in start
    _st_lottie(
```

I'll check that a bit later but if you happen to have a quick guess that may help!

**Pull request merged**